### PR TITLE
`gardenlet`'s `Seed` controller deploys MCM-related `CustomResourceDefinition`s

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area dev-productivity
/kind enhancement

**What this PR does / why we need it**:
This PR makes `gardenlet`'s `Seed` controller to manage the MCM-related `CustomResourceDefinition`s. Extensions do no longer need to take care about it when they are deployed to seed clusters with a `gardenlet` of at least `v1.73`.

The manifests were copied from https://github.com/gardener/gardener/tree/master/extensions/pkg/controller/worker/templates.

Additionally, this PR extends the `Seed` controller integration test such that it checks for the expected CRDs for both scenarios (seed is the garden; seed is not the garden). This revealed that earlier `gardenlet` was still deploying the HVPA and istio-related CRDs even when the seed was a garden. However, in this case, `gardener-operator` takes over management of these CRDs. This got fixed on the way.

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/7594

**Special notes for your reviewer**:
/cc @himanshu-kun @rishabh-11 @kon-angelo 

**Release note**:

```other developer
`gardenlet` is taking over management of the `CustomResourceDefinition`s for the `machine.sapcloud.io/v1alpha1` API group, hence extensions do no longer need to take care. Consequently, the `extensions/pkg/controller/worker.Options` struct as well as the `extensions/pkg/controller/worker.ApplyMachineResources{ForConfig}` functions are deprecated and will be removed in a future release. 
```

### CODE REVIEW

File: extensions/pkg/controller/worker/machine_crds.go [hunk 1]
```diff
@@ -71,6 +71,10 @@ func init() {
 }
 
 // ApplyMachineResourcesForConfig ensures that all well-known machine CRDs are created or updated.
+// Deprecated: This function is deprecated and will be dropped after v1.76 was released. Starting from
+// gardener/gardener@v1.73, gardenlet is managing the CRDs for the machine-controller-manager. Hence, extensions do not
+// need to take care about it anymore.
+// TODO(rfranzke): Remove this flag after v1.76 was released.
```
Comment (ScheererJ) [inline extensions/pkg/controller/worker/machine_crds.go]: Just a small nit:

```suggestion
// TODO(rfranzke): Remove this function after v1.76 was released.
```

File: extensions/pkg/controller/worker/machine_crds.go [hunk 2]
```diff
@@ -81,6 +85,10 @@ func ApplyMachineResourcesForConfig(ctx context.Context, config *rest.Config) er
 }
 
 // ApplyMachineResources ensures that all well-known machine CRDs are created or updated.
+// Deprecated: This function is deprecated and will be dropped after v1.76 was released. Starting from
+// gardener/gardener@v1.73, gardenlet is managing the CRDs for the machine-controller-manager. Hence, extensions do not
+// need to take care about it anymore.
+// TODO(rfranzke): Remove this flag after v1.76 was released.
```
Comment (ScheererJ) [inline extensions/pkg/controller/worker/machine_crds.go]: Just a small nit:

```suggestion
// TODO(rfranzke): Remove this function after v1.76 was released.
```

File: extensions/pkg/controller/worker/options.go [hunk 3]
```diff
@@ -25,6 +25,10 @@ const (
 )
 
 // Options are command line options that can be set for controller.Options.
+// Deprecated: This function is deprecated and will be dropped after v1.76 was released. Starting from
+// gardener/gardener@v1.73, gardenlet is managing the CRDs for the machine-controller-manager. Hence, extensions do not
+// need to take care about it anymore.
+// TODO(rfranzke): Remove this flag after v1.76 was released.
```
Comment (ScheererJ) [inline extensions/pkg/controller/worker/options.go]: Just a small nit:

```suggestion
// TODO(rfranzke): Remove this struct after v1.76 was released.
```

File: extensions/pkg/controller/worker/options.go [hunk 4]
```diff
@@ -25,6 +25,10 @@ const (
 )
 
 // Options are command line options that can be set for controller.Options.
+// Deprecated: This function is deprecated and will be dropped after v1.76 was released. Starting from
```
Comment (ScheererJ) [inline extensions/pkg/controller/worker/options.go]:
```suggestion
// Deprecated: This functionality is deprecated and will be dropped after v1.76 was released. Starting from
```

**Review by ScheererJ — COMMENTED** [review]
Looks good overall. Thanks for the change.

What I do not fully understand, yet, is how we ensure that the machine controller manager related custom resource definitions are only applied by one component. It looks like we deploy them now unconditionally via `gardenlet` and document that extensions do not need to take care anymore. However, if a provider extension is not updated in parallel with the inverse change the custom resource definitions could be applied by two entities. Given that these CRDs do not change all that much it might be ok. Still, I would be interested in whether this was though of during the creation of the PR.

**Comment by rfranzke** [comment]
> What I do not fully understand, yet, is how we ensure that the machine controller manager related custom resource definitions are only applied by one component. It looks like we deploy them now unconditionally via `gardenlet` and document that extensions do not need to take care anymore. However, if a provider extension is not updated in parallel with the inverse change the custom resource definitions could be applied by two entities. Given that these CRDs do not change all that much it might be ok. Still, I would be interested in whether this was though of during the creation of the PR.

You're right - we don't ensure that. The CRDs were last updated months ago and should be quite stable for the next few releases until the code is finally dropped from the extensions library. In the meantime, both `gardenlet` and provider extensions might apply them.

**Pull request merged**